usr/share/device-quirks/scripts/asus/scripts.d/ally_pipewire.sh

```diff
@@ -31,30 +37,73 @@ device_quirk_install(){
         if [ $? -eq 0 ]; then
+            echo "Pipewire configuration installed successfully."
             return 0
         else
             echo "Error copying pipewire directory into /etc"
             return 1
         fi
     fi
+
+    # Wireplumber configuration installation
+    if [[ -d "${WIREPLUMBER_DIR}" ]] && [[ -n "$(diff -qr "${WIREPLUMBER_DIR}" "${ALLY_WIREPLUMBER_DIR}")" ]]; then
+        echo "Wireplumber configurations differ. Please review the differences:"
+        diff -qr "${WIREPLUMBER_DIR}" "${ALLY_WIREPLUMBER_DIR}"
+        return 2
+    elif [[ -d "${WIREPLUMBER_DIR}" ]]; then
+        echo "Renaming existing ${WIREPLUMBER_DIR} directory to wireplumber.bak."
+        mv "${WIREPLUMBER_DIR}" "${WIREPLUMBER_DIR}.bak"
+        if [ $? -eq 0 ]; then
+            cp -r "${ALLY_WIREPLUMBER_DIR}" "${DQ_WORKING_PATH}/etc/"
+            if [[ ! -f "${WIREPLUMBER_DIR}/.ALLY-WIREPLUMBER-CFG" ]]; then
+                touch "${WIREPLUMBER_DIR}/.ALLY-WIREPLUMBER-CFG"
+            fi
+            echo "Wireplumber configuration installed successfully."
+            return 0
+        else
+            echo "Error renaming existing wireplumber directory."
+            return 1
+        fi
+    else
+        cp -r "${ALLY_WIREPLUMBER_DIR}" "${DQ_WORKING_PATH}/etc/"
+        if [ $? -eq 0 ]; then
+            echo "Wireplumber configuration installed successfully."
+            return 0
+        else
+            echo "Error copying wireplumber directory into /etc"
+            return 1
+        fi
+    fi
 }
 
-# Remove the install here.
 device_quirk_removal(){
 
-    if [[ -d "${PIPEWIRE_DIR}" ]] && [[ -z $(diff -qr "${PIPEWIRE_DIR}" "${ALLY_PIPEWIRE_DIR") ]]; then
+    # Pipewire configuration removal
+    if [[ -d "${PIPEWIRE_DIR}" ]] && [[ -z "$(diff -qr "${PIPEWIRE_DIR}" "${ALLY_PIPEWIRE_DIR}")" ]]; then
         rm -rf "${DQ_WORKING_PATH}/etc/pipewire"
         if [ $? -eq 0 ]; then
-            return 0
+            echo "Successfully removed pipewire directory."
         else
             echo "Error removing pipewire directory."
             return 1
         fi
-    elif [[ -d "${PIPEWIRE_DIR}" ]]; then
-        echo "Manual removal of the Asus ROG Ally pipewire configuration required..."
-        return 1
     else
-        return 2
+        echo "Pipewire configuration not found or differs from the expected configuration."
+    fi
+
+    # Wireplumber configuration removal
+    if [[ -d "${WIREPLUMBER_DIR}" ]] && [[ -z "$(diff -qr "${WIREPLUMBER_DIR}" "${ALLY_WIREPLUMBER_DIR}")" ]]; then
+        rm -rf "${DQ_WORKING_PATH}/etc/wireplumber"
+        if [ $? -eq 0 ]; then
+            echo "Successfully removed wireplumber directory."
+            return 0
+        else
+            echo "Error removing wireplumber directory."
+            return 1
+        fi
+    else
+        echo "Wireplumber configuration not found or differs from the expected configuration."
```

Comment:
same for the return 2 here. Since the only other case always ends in a return we can just remove the else and put the echo outside the if statement